### PR TITLE
Version 256

### PR DESCRIPTION
**Pull request by djarek**
* Remove uses of the deprecated `buffers` function
* Remove uses of deprecated methods in websocket tests
* Remove redundant use of `static_string`
* Remove redundant template in service_base
* Expand CI matrix using Azure Pipelines
* Make chat websocket javascript client more user friendly
* `allocator_traits::construct` is used for user-defined types
* Add 1-element specialization for `buffers_cat`
* Fix `buffers_cat` iterator tests
* Don't pessimize-move
* Use steady_timer type
* Preserve operation_aborted on partial message

### CODE REVIEW

File: include/boost/beast/core/impl/basic_stream.hpp [hunk 1]
```diff
@@ -161,12 +160,21 @@ close()
 //------------------------------------------------------------------------------
 
 template<class Protocol, class Executor, class RatePolicy>
+template<class Executor2>
```
Comment (vinniefalco) [inline include/boost/beast/core/impl/basic_stream.hpp]: I'm not generally opposed to this change, but aren't we just trading one instantiation for another?

File: include/boost/beast/core/detail/service_base.hpp [hunk 2]
```diff
@@ -16,15 +16,10 @@ namespace boost {
 namespace beast {
 namespace detail {
 
-template<class T>
-struct service_id : net::execution_context::id
```
Comment (vinniefalco) [inline include/boost/beast/core/detail/service_base.hpp]: Doesn't asio do it this way? I thought it was a workaround for a link problem... are you sure this change will be correct everywhere?

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,7 @@ Version 256:
 * Remove redundant template in service_base
 * Expand CI matrix using Azure Pipelines
 * Make chat websocket javascript client more user friendly
+* Don't call `allocator_traits<T>::construct` unnecessarily
```
Comment (vinniefalco) [inline CHANGELOG.md]: The change message could be "allocator_traits<T>::construct is for user-defined types"

File: include/boost/beast/core/impl/basic_stream.hpp (hunk 1 above)
Comment (djarek) [reply include/boost/beast/core/impl/basic_stream.hpp]: `bind_executor` has to instantiate at least 3 additional classes (due to support for binding executors for CompletionTokens) and adds a few hundred lines of header material.

File: include/boost/beast/core/detail/service_base.hpp (hunk 2 above)
Comment (djarek) [reply include/boost/beast/core/detail/service_base.hpp]: It's possibly a workaround for a problem in a toolset that is not supported by beast anyway.

**Pull request merged**